storage/tests/unit/test__helpers.py

```diff
@@ -36,15 +36,26 @@ class Derived(self._get_target_class()):
             def path(self):
                 return path
 
+            @property
+            def user_project(self):
+                return user_project
+
         return Derived
 
     def test_path_is_abstract(self):
         mixin = self._make_one()
-        self.assertRaises(NotImplementedError, lambda: mixin.path)
+        with self.assertRaises(NotImplementedError):
+            mixin.path
```

Comment:
`nox -e lint` doesn't barf there.  It seems clear to me what is happening, and adding the `getattr` doesn't make it any clearer (lint might barf at not using the return value ;)